src/views/open-api-docs-view/components/open-api-operations/index.tsx

```diff
@@ -37,7 +38,13 @@ export function OpenApiOperations({
 							key={operation.operationId}
 							style={{ border: '1px solid magenta', padding: '2px' }}
 						>
-							<OperationHeader operation={operation} />
+							<OperationHeader
+								className={s.header}
+								slug={operation.slug}
+								headerText={operation.operationId}
+								method={operation.type}
+								path={operation.path.truncated}
```

Comment:
⛏️ Okay this is a weird one, and I think is also a problem upstream, but long `path` and also long `headerText` strings can cause overflow and wreak havok on the layout on small viewports, cause they tend to not have any spaces or word-break-able characters:

| overflow issue |
| - |
| <img src="https://github.com/hashicorp/dev-portal/assets/4624598/01ab973c-fc66-4186-aa1f-89f49b7a9ef5" alt="overflow-issue" width="320" /> |

A "quick fix" could be `overflow-wrap: break-word`, I think 🤔 A more involved fix might be to get `<wbr>` elements interspersed into `headerText` and `path` somehow, which might yield a more polished result:

| `overflow-wrap` | `<wbr>` |
| - | - |
| <img width="512" alt="overflow-wrap-anywhere" src="https://github.com/hashicorp/dev-portal/assets/4624598/0f9030cf-51cd-4676-9ed5-13b27da46540"> | <img width="2098" alt="injected-word-breaks" src="https://github.com/hashicorp/dev-portal/assets/4624598/bd03180e-46d4-4bd3-919f-914083e7f54b"> |

I think we'll actually want `overflow-wrap: break-word` even if we go for the more involved `<wbr>` addition, since we still won't be able to guarantee we've added `<wbr>` in enough spots... so that's something at least! I'd definitely say we could roll with that CSS only fix for now 👍

---

... that being said but I can't help jumping to the conclusion that we might want to pursue the `<wbr>` approach, so I noodled on that over in a branch off this PR:

- https://github.com/hashicorp/dev-portal/pull/2103

Reply:
I love your exploration with the word-breaks and the idea that we could life that into a higher utility. I'll push up the CSS only fix for now so we can play around more with the work break approach.